### DISCUSSION TITLE
SORT BY COLUMN does not work, when the "result-fetch size" is less than the number of records

### DISCUSSION THREAD

**Discussion by danilort** [dbeaver/dbeaver#12050] [Q&A]
#### System information: 
- Operating system (distribution) and version: Windows 10 versione 20H2 build 19042.804
- DBeaver version: DBeaver Community 21.0.0.202103021012	
- Additional extensions: NULL

#### Connection specification:
- Database name and version: MySQL 8.0.21
- Driver name: Driver for MySQL 8 and later
- Do you use tunnels or proxies (SSH, SOCKS, etc)? NULL

#### Describe the problem you're observing:
SORT BY COLUMN does not work, when the "result-fetch size" is less than the number of records

#### Steps to reproduce, if exist:

1) SET result-fetch size = **100**   (less than the number of record)
2) Run the query. In the SQL Console I see: 
SELECT * FROM vaccessi WHERE dataora > '2020-03-07 09:18:00' ORDER BY id ASC 
3) CLIC BUTTON: "sort by column" on column "metodo". In the SQL Console I see:
SELECT * FROM vaccessi WHERE dataora > '2020-03-07 09:18:00' **ORDER BY id ASC, metodo**

RESULT: the column "metodo" **is not sorted**

In this way it works well:
1) SET result-fetch size = **10000**   (greater than the number of record)
2) Run the query. In the SQL Console I see:
SELECT * FROM vaccessi WHERE dataora > '2020-03-07 09:18:00' ORDER BY id ASC 
3) CLIC BUTTON: "sort by column" on column "metodo". In the SQL Console I see:
SELECT * FROM vaccessi WHERE dataora > '2020-03-07 09:18:00' **ORDER BY id ASC** 

RESULT: the column "metodo" **is sorted**

NOTE: The two #3 are different

**Comment by uslss** [comment]
hi @danilort ,
unfortunately cannot reproduce it. Could you provide table ddl?

**Comment by danilort** [comment]
The problem also occurs using SQLite.
I send you the database dgshopping_temp_32.db

Query:  SELECT * FROM oggetti ORDER BY id ASC;
See screenshot.jpg

![immagine](https://user-images.githubusercontent.com/16148184/110448823-d8d7d900-80c1-11eb-94d5-9ade2e5c3ca4.png)


[dgshopping_temp_32.zip](https://github.com/dbeaver/dbeaver/files/6107406/dgshopping_temp_32.zip)

**Comment by uslss** [comment]
thanks for clarification

**Comment by LonwoLonwo** [comment]
Hello @danilort 

First of all in this case:
`SELECT * FROM oggetti ORDER BY id ASC, nome`
The values of the 'nome' column will not be sorted, because sorting the second column from the order depends on the grouping of the values of the first specified column in ORDER BY case - which is not grouped, because it is id.

As example.

id    |     Name
---------------
1     |     Willy
3     |     Alla
2     |     Nancy
4     |     Sam

`SELECT * FROM table ORDER BY id ASC, Name` returns:

id    |     Name
---------------
1     |     Willy
2     |     Nancy
3     |     Alla
4     |     Sam

But in this case:

Number  |     Name
---------------------
1            |     Willy
2            |     Mary
4            |     Sam
4            |     Andy
2            |     Nancy
1            |     Alla

`SELECT * FROM table ORDER BY Number ASC, Name` query returns:

Number  |     Name
---------------------
1            |     Alla
1            |     Willy
2            |     Mary
2            |     Nancy
4            |     Andy
4            |     Sam

So, the question now is - why DBeaver works in your second case and does not work in first?
DBeaver caches values from table.  And in second case DBeaver caches all values and can sort from client side.
In first case DBeaver do not have all values from table because their number is limited. So then he request SELECT query with ORDER clauses to server-side. 

If you want to sort by column 'nome' - just do not sort on id column. Or use some DBeaver workaround from your second example :)
